code/modules/mob/living/carbon/human/species/unathi.dm

```diff
@@ -131,12 +131,15 @@
 	var/datum/action/innate/ignite/ash_walker/fire = new()
 	fire.Grant(H)
 	RegisterSignal(H, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(speedylegs))
+	if(!is_mining_level(H.z))
+		speed_mod = 0
 
 /datum/species/unathi/ashwalker/on_species_loss(mob/living/carbon/human/H)
 	..()
 	for(var/datum/action/innate/ignite/ash_walker/fire in H.actions)
 		fire.Remove(H)
 	UnregisterSignal(H, COMSIG_MOVABLE_Z_CHANGED)
+	speed_mod = 0
```

Comment:
This shouldnt ever be needed, because this species is getting deleted

Reply:
This is in case a cling transforms from Ash Walker to something else.

Reply:
We're losing this species, and this is a var on the species. It won't do anything if you set it to zero, so its best to not have it.